webboot/webboot.go

```diff
@@ -9,60 +9,146 @@ package main
 	"github.com/u-root/webboot/pkg/mountkexec"
 	"github.com/u-root/webboot/pkg/webboot"
 )
 
 var (
-	cmd       = flag.String("cmd", "", "Command Line")
-	mountDir  = flag.String("dir", "/tmp/mountDir", "The mount point of the ISO")
-	ifName    = flag.String("interface", "^e.*", "Name of the interface")
-	timeout   = flag.Int("timeout", 15, "Lease timeout in seconds")
-	retry     = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
-	verbose   = flag.Bool("verbose", false, "Verbose output")
-	ipv4      = flag.Bool("ipv4", true, "use IPV4")
-	ipv6      = flag.Bool("ipv6", true, "use IPV6")
-	osSystems = map[string]*webboot.Distro{
-		"tinycore": &webboot.Distro{"/boot/vmlinuz64", "/boot/corepure64.gz", "console=tty1", "https:louis.com"},
-		"TestISO":  &webboot.Distro{"/kernel", "/initiso.cpi", "console=ttyS0 console=tty0", "https:louis.com"},
+	cmd      = flag.String("cmd", "", "Command Line")
+	mountDir = flag.String("dir", "/tmp/mountDir", "The mount point of the ISO")
+	ifName   = flag.String("interface", "^e.*", "Name of the interface")
+	timeout  = flag.Int("timeout", 15, "Lease timeout in seconds")
+	retry    = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
+	verbose  = flag.Bool("verbose", false, "Verbose output")
+	ipv4     = flag.Bool("ipv4", true, "use IPV4")
+	ipv6     = flag.Bool("ipv6", true, "use IPV6")
+	bookmark = map[string]*webboot.Distro{
+		"tinycore": &webboot.Distro{"/boot/vmlinuz64", "/boot/corepure64.gz", "console=tty1", "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso"},
 	}
 )
 
+// parseArg takes a name and produces a filename and a URL
+// The URL can be used to download data to the file 'filename'
+// The argument is either a full URL or a bookmark.
+func parseArg(arg string) (string, string, error) {
+	var URL, filename string
+	if u, ok := bookmark[arg]; ok {
+		return u.DownloadLink, arg, nil
+	}
+
+	var err error
+	filename, err = name(arg)
+	URL = arg
+	if err != nil {
+		return "", "", fmt.Errorf("%v is not a valid URL: %v", URL, err)
+	}
+
+	return URL, filename, nil
+}
+
+// linkOpen returns an io.ReadCloser that holds the content of the URL
+func linkOpen(URL string) (io.ReadCloser, error) {
+	resp, err := http.Get(URL)
+	if err != nil {
+		return nil, err
+	}
+	// checks if request to receive URL is successful
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+// write copies the content from an io.readCloser to a named file with filename.
+func write(read io.ReadCloser, filename string) error {
+	w, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+
+	defer w.Close()
+	defer read.Close()
+
+	if _, err := io.Copy(w, read); err != nil {
+		return fmt.Errorf("Error copying %v: %v", filename, err)
+	}
+	return nil
+}
+
+// name takes a URL & generates a filename from it
```

Comment:
Please don't use & in these comments.

Reply:
Sorry about that, I changed it to 'and', and got rid of "&" :)